### PR TITLE
Feature/edit content tools

### PR DESCRIPTION
**Pull request by vxsx**
forgot to add setup/teardown in editContentTools + additional fixes

### CODE REVIEW

File: cms/tests/frontend/integration/editContentTools.js [hunk 1]
```diff
@@ -107,7 +156,26 @@ casper.test.begin('Edit utils page content', function (test) {
...
+            test.assertElementCount(
+                '.cms-structure .cms-draggables .cms-draggable',
+                contentNumber,
+                messages.pasteAfterCutSuccessful
+            );
+        })
+
+        // wait till the paste actually succeeds
+        .waitForResource(/move\-plugin/)
+        .wait(1000)
```
Comment (FinalAngel) [inline cms/tests/frontend/integration/editContentTools.js]: do we really need hardcoded timeouts? Can't that fire back at us?

File: cms/tests/frontend/integration/helpers/cms.js [hunk 2]
```diff
@@ -129,6 +129,16 @@ module.exports = function (casperjs) {
                         this.click('.cms-modal-buttons .cms-btn-action.default');
                     });
             };
+        },
+
+        clearClipboard: function () {
+            return function () {
```
Comment (FinalAngel) [inline cms/tests/frontend/integration/helpers/cms.js]: I don't understand the returns, why do we need a self executing function?

File: cms/tests/frontend/integration/editContentTools.js (hunk 1 above)
Comment (vxsx) [inline cms/tests/frontend/integration/editContentTools.js]: we kinda do :) i tried to make it small enough
sadly `waitForResource` is somewhat unreliably reporting that it succeeded, but i see in the logs that it did not, so yeah, wait at least 1 second before rechecking

File: cms/tests/frontend/integration/helpers/cms.js (hunk 2 above)
Comment (vxsx) [inline cms/tests/frontend/integration/helpers/cms.js]: if you look at the usage:
```js
casper.start()
        .then(cms.clearClipboard())
        .then(cms.removePage())
        .then(cms.logout())
        .run(done);
```
Casper's `.then()` accepts a function, so in order to pass it a function i have to return a function. I could just do
`casper.start().then(cms.clearClipboard).then(run)` to pass a function reference but sometimes we need to pass arguments to the helper (page title, plugin contents etc) so i decided to make them consistently callable

work similar to gulp plugins, where the plugin returns a function that is then run on the stream's data

**Comment by FinalAngel** [comment]
Thanks for the explanation - LGTM

**Pull request merged**